crates/foundry/evm/core/src/opts.rs

```diff
@@ -329,6 +330,24 @@ impl From<TxEnvOpts> for TxEnv {
     }
 }
 
+impl From<TxEnvOpts> for OpTransaction<TxEnv> {
+    fn from(value: TxEnvOpts) -> Self {
+        let base = TxEnv::from(value);
+
+        OpTransaction {
+            base,
+            // We don't know enough information to determine whether the transaction is enveloped,
+            // but all transactions except deposits require this parameter, while the
+            // deposit transaction ignores it. As such, we're always setting this to a default
+            // value.
+            // TODO: https://github.com/NomicFoundation/edr/issues/934
+            // TODO: Properly calculate `enveloped_tx` to ensure correct gas calculation.
```

Comment:
See comments.

I think it's fine to leave this out of scope for the current version of multi-chain Solidity tests, but we should have a discussion about whether we want to support this in the future and how.